packages/wallet/src/services/TransactionReemitter.ts

```diff
@@ -0,0 +1,66 @@
+import { Cardano } from '@cardano-sdk/core';
+import { DocumentStore } from '../persistence';
+import { NewTxAlonzoWithSlot } from './types';
+import { Observable, filter, from, map, merge, mergeMap, scan, withLatestFrom } from 'rxjs';
+
+interface TransactionReemitterProps {
+  rollback$: Observable<Cardano.TxAlonzo>;
+  confirmed$: Observable<NewTxAlonzoWithSlot>;
+  submitting$: Observable<Cardano.NewTxAlonzo>;
+  store: DocumentStore<NewTxAlonzoWithSlot[]>;
+  stabilityWindowSlotsCount?: number;
+}
+
+enum txSource {
+  store,
+  confirmed,
+  submitting
+}
+
+// 3k/f (where k is the security parameter in genesis, and f is the active slot co-efficient parameter
+// in genesis that determines the probability for amount of blocks created in an epoch.)
+const kStabilityWindowSlotsCount = 129_600; // 3k/f on current mainnet
+
+export const createTransactionReemitter = ({
+  rollback$,
+  confirmed$,
+  submitting$,
+  store,
+  stabilityWindowSlotsCount = kStabilityWindowSlotsCount
+}: TransactionReemitterProps): Observable<Cardano.NewTxAlonzo> => {
+  const volatileTransactions$ = merge(
+    store.get().pipe(
+      mergeMap((txs) => from(txs)),
+      map((tx) => ({ source: txSource.store, tx }))
+    ),
+    confirmed$.pipe(map((tx) => ({ source: txSource.confirmed, tx }))),
+    submitting$.pipe(map((tx) => ({ source: txSource.submitting, tx: { ...tx, slot: null! } })))
+  ).pipe(
+    scan((volatiles, { tx, source }) => {
+      switch (source) {
+        case txSource.store:
+          // Do not calculate stability window for old transactions coming from the store
+          return [...volatiles, tx];
+        case txSource.submitting:
+          // Transactions in submitting are the ones reemitted. Remove them from volatiles
+          return volatiles.filter((v) => v.id !== tx.id);
+        case txSource.confirmed: {
+          const oldestAcceptedSlot = tx.slot > stabilityWindowSlotsCount ? tx.slot > stabilityWindowSlotsCount : 0;
+          // Remove transactions considered stable
+          return [...volatiles.filter(({ slot }) => slot > oldestAcceptedSlot), tx];
+        }
+      }
+    }, [] as NewTxAlonzoWithSlot[])
+  );
+
+  return rollback$.pipe(
+    withLatestFrom(volatileTransactions$),
+    map(
+      ([{ id }, volatiles]) =>
+        // REVIEW: all examples of logger imply passing it as parameter. Is that the only way to do it?
```

Comment:
Logger question :)

Reply:
We've not really established a pattern for logging within the wallet package. Do you see an issue with passing it through this way?

Reply:
No. It's just that I have add one more param and I wasn't sure if there is a better way to do it.
I will do that